.github/workflows/build_and_test.yml

```
@@ -35,220 +35,234 @@ on:
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.aws-access-key-id }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.aws-secret-access-key  }}
+          GOOGLE_CREDENTIALS_PATH: "/home/runner/work/nuxeo-ai/nuxeo-ai/gcp-creds.json"
+      - name: Publish Test Report
+        if: ${{ always() }}
+        uses: scacap/action-surefire-report@v1
+      - name: Get project version
+        run: echo "VERSION=$(mvn -q -Dexec.executable=echo -Dexec.args='${project.version}' --non-recursive exec:exec | xargs echo -n)" >> $GITHUB_ENV
+      - run: ls -alR | grep ".*package.*.zip"
+      - run: ls -alR | grep ".*nuxeo-ai-core-.*.zip"
+      - name: Upload build artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: nuxeo-ai-github-build
+          path: |
+            ./**/target/**/*package*.zip
+            ./nuxeo-ai-core-package/target/nuxeo-ai-core-*.zip
+      - name: Publish package
+        if: ${{ inputs.should-push }}
+        run: mvn --batch-mode deploy -nsu -DskipTests
+      - name: Publish nuxeo-ai-core package to Production Connect Marketplace
+        if: ${{ inputs.should-push }}
+        env:
+          PACKAGE: nuxeo-ai-core-package/target/nuxeo-ai-core-${{ env.VERSION }}.zip
+          CONNECT_URL: https://connect.nuxeo.com/nuxeo
+        run: curl -i -u "${{ secrets.connect-auth-user }}:${{ secrets.connect-auth-token }}" -F package=@$PACKAGE "$CONNECT_URL/site/marketplace/upload?batch=true"
+      - name: Publish nuxeo-ai-aws-package package to Production Connect Marketplace
+        if: ${{ inputs.should-push }}
+        env:
+          PACKAGE: addons/nuxeo-ai-aws-package/target/nuxeo-ai-aws-package-${{ env.VERSION }}.zip
+          CONNECT_URL: https://connect.nuxeo.com/nuxeo
+        run: curl -i -u "${{ secrets.connect-auth-user }}:${{ secrets.connect-auth-token }}" -F package=@$PACKAGE "$CONNECT_URL/site/marketplace/upload?batch=true"
+      - name: Publish nuxeo-ai-gcp-package package to Production Connect Marketplace
+        if: ${{ inputs.should-push }}
+        env:
+          PACKAGE: addons/nuxeo-ai-gcp-package/target/nuxeo-ai-gcp-package-${{ env.VERSION }}.zip
+          CONNECT_URL: https://connect.nuxeo.com/nuxeo
+        run: curl -i -u "${{ secrets.connect-auth-user }}:${{ secrets.connect-auth-token }}" -F package=@$PACKAGE "$CONNECT_URL/site/marketplace/upload?batch=true"
+      - name: Publish nuxeo-ai-image-quality-package package to Production Connect Marketplace
+        if: ${{ inputs.should-push }}
+        env:
+          PACKAGE: addons/nuxeo-ai-image-quality-package/target/nuxeo-ai-image-quality-package-${{ env.VERSION }}.zip
+          CONNECT_URL: https://connect.nuxeo.com/nuxeo
+        run: curl -i -u "${{ secrets.connect-auth-user }}:${{ secrets.connect-auth-token }}" -F package=@$PACKAGE "$CONNECT_URL/site/marketplace/upload?batch=true"
+
+  notify-on-error:
+    runs-on: ubuntu-latest
+    needs: [build, build-master]
+    if: ${{ failure() }}
```

Since the "CI for PR" (`pipline_pr.yaml`) workflow calls this workflow for pull requests and pushes, this could alert on pull requests now.

https://github.com/nuxeo/nuxeo-ai/blob/63930dbee31bd633c43c4b1eca91802619888bca/.github/workflows/pipeline_pr.yaml#L1-L6

https://github.com/nuxeo/nuxeo-ai/blob/63930dbee31bd633c43c4b1eca91802619888bca/.github/workflows/pipeline_pr.yaml#L9-L14

Could you update this condition to also check for the `should-push` input or as an alternative, could you move this job to the `pipeline_pr` workflow and check that the event is "push"?

```suggestion
    if: ${{ failure() && inputs.should-push }}
```

https://github.com/nuxeo/nuxeo-ai/blob/8fb7348bfc2bd715eab99050df7f1de378b737f9/.github/workflows/build_and_test.yml#L16-L19

Either of those two changes would make it so pull request workflow failures wouldn't notify our team while we would get notifications of release failures.